cmd/api/src/daemons/datapipe/jobs.go

```diff
@@ -119,26 +121,69 @@ func (s *Daemon) clearFileTask(ingestTask model.IngestTask) {
 	}
 }
 
-func (s *Daemon) processIngestFile(ctx context.Context, path string) error {
-	if jsonFile, err := os.Open(path); err != nil {
-		return err
+// preProcessIngestFile will take a path and extract zips if necessary, returning the paths for files to process
+func (s *Daemon) preProcessIngestFile(path string, fileType model.FileType) ([]string, error) {
+	if fileType == model.FileTypeJson {
+		//If this isn't a zip file, just return a slice with the path in it and let stuff process as normal
+		return []string{path}, nil
+	} else if archive, err := zip.OpenReader(path); err != nil {
+		return []string{}, err
 	} else {
-		defer func() {
-			if err := jsonFile.Close(); err != nil {
-				log.Errorf("Failed closing ingest file %s: %v", path, err)
+		filePaths := make([]string, len(archive.File))
+		for i, f := range archive.File {
+			//skip directories
+			if f.FileInfo().IsDir() {
+				continue
 			}
-		}()
-
-		return s.graphdb.BatchOperation(ctx, func(batch graph.Batch) error {
-			if err := ReadFileForIngest(batch, jsonFile); err != nil {
-				return err
+			//Break out on an error creating files
+			if tempFile, err := os.CreateTemp(s.cfg.TempDirectory(), "bh"); err != nil {
+				return []string{}, err
+			} else if srcFile, err := f.Open(); err != nil {
+				log.Errorf("Error opening file %s in archive %s: %v", f.Name, path, err)
+			} else if _, err := io.Copy(tempFile, srcFile); err != nil {
+				log.Errorf("Error extracting file %s in archive %s: %v", f.Name, path, err)
+			} else if err := tempFile.Close(); err != nil {
+				log.Errorf("Error closing temp file %s: %v", f.Name, err)
 			} else {
-				return nil
+				filePaths[i] = tempFile.Name()
 			}
-		})
+		}
+
+		//Close the archive and delete it
+		if err := archive.Close(); err != nil {
+			log.Errorf("Error closing archive %s: %v", path, err)
+		} else if err := os.Remove(path); err != nil {
+			log.Errorf("Error deleting archive %s: %v", path, err)
```

Comment:
This is probably correct behavior but just confirming that we want to delete archives even if they partially fail to process.